pkg/policy/identity.go

```diff
@@ -191,17 +191,22 @@ const (
 	// ReservedIdentityCluster represents any endpoint inside the cluster
 	// that does not have a more specific identity
 	ReservedIdentityCluster
+
+	// ReservedIdentityHealth represents the local cilium-health endpoint
+	ReservedIdentityHealth
```

Comment:
These endpoints are intended to be reserved for cilium debug use, and when writing this tool I've been thinking about how to answer questions like:

* Does basic node-to-node connectivity work? (already merged)
* Does basic container-to-container connectivity work? (this commit)
* Does connectivity work when there's an allow policy in place? (future)
* Does connectivity work when packets are sent through a proxy? (future)

For the last two items above, I expect that we'll want to reference these endpoints in policies. However, I'd expect these to be managed by cilium and not by end users.

The defaults of this tool are set up with the idea in mind that you don't have to turn something on, or deploy something, etc. when you're hitting a connectivity issue. The daemons are running, you just need to query them and interpret the output.

With regards to users wanting to explicitly blocking health, I think that a configuration option on the daemon would be the best option to address this, so we don't spawn these extra daemons on startup. This shouldn't be too much work, but at this stage I plan to defer it until it is requested.

Reply:
> For the last two items above, I expect that we'll want to reference these endpoints in policies. However, I'd expect these to be managed by cilium and not by end users.

Yeah, I was mostly thinking along the lines of not having special cases in the internal policy resolution. We'd deny user policies from having one thing or the other during validation. I'm not sure how much work this would create, though.

 > With regards to users wanting to explicitly blocking health, I think that a configuration option on the daemon would be the best option to address this, so we don't spawn these extra daemons on startup. This shouldn't be too much work, but at this stage I plan to defer it until it is requested.

How would that manifest in a health probe? if nothing is listening the connection would be refused. Would the report distinguish in some way? Since this option would be an agent level thing, but it may not correlate with what pods are scheduled on that node it might be a little confusing? I haven't thought through it.

Reply:
>> With regards to users wanting to explicitly blocking health, I think that a configuration option on the daemon would be the best option to address this, so we don't spawn these extra daemons on startup. This shouldn't be too much work, but at this stage I plan to defer it until it is requested.

> How would that manifest in a health probe? if nothing is listening the connection would be refused. Would the report distinguish in some way? Since this option would be an agent level thing, but it may not correlate with what pods are scheduled on that node it might be a little confusing? I haven't thought through it.

I'd expect that remote instances that still have health enabled would observe that the "health-disabled" node exists and try to ping it, but the ping would fail. If you wanted to distinguish deliberate configuration vs. unexpected connectivity loss, the "health-disabled" instance would have to propagate the fact that its health is deliberately disabled to the other cilium instances, expose it to cilium-health, which could then print the output differently for those nodes.